src/commands/generate_buildpack_matrix/command.rs

```diff
@@ -19,41 +20,55 @@ type Result<T> = std::result::Result<T, Error>;
 #[command(author, version, about = "Generates a JSON list of buildpack information for each buildpack detected", long_about = None)]
 pub(crate) struct GenerateBuildpackMatrixArgs {
     #[arg(long)]
-    pub(crate) package_dir: PathBuf,
-    #[arg(long, default_value = "x86_64-unknown-linux-musl")]
-    pub(crate) target: String,
+    pub(crate) source_dir: Option<PathBuf>,
+    #[arg(long)]
+    pub(crate) package_dir: Option<PathBuf>,
+    #[arg(long)]
+    pub(crate) temporary_id: String,
 }
 
 pub(crate) fn execute(args: &GenerateBuildpackMatrixArgs) -> Result<()> {
-    let current_dir = std::env::current_dir().map_err(Error::GetCurrentDir)?;
-    let package_dir = resolve_path(&args.package_dir, &current_dir);
-
-    let packaged_buildpack_dir_resolver =
-        create_packaged_buildpack_dir_resolver(&package_dir, CargoProfile::Release, &args.target);
+    let source_dir = match &args.source_dir {
+        Some(path) => path.clone(),
+        None => std::env::current_dir().map_err(Error::GetCurrentDir)?,
+    };
+    let package_dir = resolve_path(
+        match &args.package_dir {
+            Some(path) => path,
+            None => Path::new("./packaged"),
+        },
+        &source_dir,
+    );
 
     let buildpack_dirs =
-        find_releasable_buildpacks(&current_dir).map_err(Error::FindReleasableBuildpacks)?;
+        find_releasable_buildpacks(&source_dir).map_err(Error::FindReleasableBuildpacks)?;
 
     let buildpacks = buildpack_dirs
         .iter()
         .map(|dir| read_buildpack_descriptor(dir).map_err(Error::ReadBuildpackDescriptor))
         .collect::<Result<Vec<_>>>()?;
 
-    let includes = buildpack_dirs
+    let buildpacks_info = buildpack_dirs
         .iter()
         .zip(buildpacks.iter())
         .map(|(buildpack_dir, buildpack_descriptor)| {
-            extract_buildpack_info(
+            read_buildpack_info(
                 buildpack_descriptor,
                 buildpack_dir,
-                &packaged_buildpack_dir_resolver,
+                &package_dir,
+                &args.temporary_id,
             )
         })
         .collect::<Result<Vec<_>>>()?;
 
-    let includes_json = serde_json::to_string(&includes).map_err(Error::SerializingJson)?;
+    let buildpacks_json =
+        serde_json::to_string_pretty(&buildpacks_info).map_err(Error::SerializingJson)?;
 
-    actions::set_output("buildpacks", includes_json).map_err(Error::SetActionOutput)?;
+    actions::set_output("buildpacks", &buildpacks_json).map_err(Error::WriteActionData)?;
+    actions::set_summary(format!(
+        "<details><summary>Buildpack Matrix</summary>\n\n```json\n{buildpacks_json}\n```\n</details>"
```

Comment:
nice to get visibility into this data finally! and good call putting this into a summary table. the output for https://github.com/heroku/buildpacks-nodejs/actions/runs/8926879828 is huge 😅 

in the future, a flattened HTML table might be a reasonable UX improvement but let's not worry about that now.

Reply:
Yeah, it's huge. Obviously, the Node.js CNB will be huge since it includes so many buildpacks, and 2x targets for each. Other buildpacks should be a little easier to parse.